### PR TITLE
Changes padding for first five pooling layers TF OD

### LINKED ISSUE #2616: OD TensorFlow model should use the same padding as 5.8
The current TensorFlow model fix the padding to `SAME` for all pooling layers,
however for 5.8 implementation, the first five pooling layers use `VALID` padding instead.

### PR DESCRIPTION
**Pull request by shreyajain17**
fixes #2616

Passes all unit tests after the change (python and cpp)

### CODE REVIEW

File: src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py [hunk 1]
```diff
@@ -262,8 +262,10 @@ def tiny_yolo(self, inputs, output_size=125):
                     strides = [1, 2, 2, 1]
                 else:
                     strides = [1, 1, 1, 1]
-
-                net = self.pooling_layer(net, pool_size=[1, 2, 2, 1], strides=strides, name='pool%d_' % idx)
+                if idx < 6 :
```
Comment (jakesabathia2) [inline src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py]: Can we only use one `if else`? Set both `stride` and `padding` in the same `if else`.

**Comment by jakesabathia2** [comment]
Oh and for cpp, the sixth pooling layer has an argument `set_avgpoolexcludepadding = true`, which might suggests that this layer is average pooling layer instead of max pooling layer:
https://github.com/apple/coremltools/blob/7f2b314f496072320e1088ad8c6f6f572063c454/coremltools/models/neural_network/builder.py#L1824
So maybe we want to use `tf.nn.avg_pooling` here instead @nickjong .

**Comment by shreyajain17** [comment]
@jakesabathia2 Good catch. I will make changes to accommodate it.

File: src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py (hunk 1 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py]: Yeah makes sense.
